zebra/redistribute.c

```diff
@@ -176,10 +176,13 @@ static bool zebra_redistribute_check(const struct route_entry *re,
 	 * If multi-instance then check for route
 	 * redistribution for given instance.
 	 */
-	if (re->instance
-	    && redist_check_instance(&client->mi_redist[afi][re->type],
-				     re->instance))
-		return true;
+	if (re->instance) {
+		if (redist_check_instance(&client->mi_redist[afi][re->type],
+					  re->instance))
+			return true;
+		else
+			return false;
```

Comment:
What if we have this configuration "router ospf 3 vrf abc", should we match both instance id and vrf id in that case?

Reply:
instance is only meaningful in the default vrf.  And you cannot create a instance based ospf vrf.